code/modules/mob/mob.dm

```diff
@@ -1111,19 +1111,43 @@
 		if(E.mouse_pointer)
 			client.mouse_pointer_icon = E.mouse_pointer
 
+/**
+ * Can this mob see in the dark
+ *
+ * This checks all traits, glasses, and robotic eyeball implants to see if the mob can see in the dark
+ * this does NOT check if the mob is missing it's eyeballs. Also see_in_dark is a BYOND mob var (that defaults to 2)
+**/
+/mob/proc/has_nightvision()
+	return see_in_dark >= NIGHTVISION_FOV_RANGE
 
-///This mob is abile to read books
+/// Can this nerd read
 /mob/proc/is_literate()
 	return FALSE
 
+/**
+ * Checks if there is enough light where the mob is located
+ *
+ * Args:
+ *  light_amount (optional) - A decimal amount between 1.0 through 0.0 (default is 0.2)
+**/
+/mob/proc/has_light_nearby(light_amount = LIGHTING_TILE_IS_DARK)
```

Comment:
seems a badly named proc "nearby" implies within tiles around the mob. Should be has_light_on_turf or something

Reply:
> has_light_on_turf

Probably will do that. I can't think of a proc name for this that doesnt sound like ass tbh